cycletls/tests/integration/client2_test.go

```diff
@@ -0,0 +1,150 @@
+package cycletls
+
+import (
+	"github.com/Danny-Dasilva/CycleTLS/cycletls"
+	http "github.com/Danny-Dasilva/fhttp"
+	utls "github.com/refraction-networking/utls"
+	"io"
+	"sync"
+	"testing"
+	"time"
+)
+
+const UserAgent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36 OPR/48.0.2685.52"
+
+func TestHttpClientBuilderWithProxyVisitingGoogle(t *testing.T) {
+
+	browser := &cycletls.Browser{
+		JA3:                "",
+		UserAgent:          UserAgent,
+		InsecureSkipVerify: true,
+	}
+
+	builder := cycletls.HttpClientBuilder{
+		Browser:       browser,
+		ClientHelloId: &utls.HelloRandomized,
+		// ProxyUrl:             "http://localhost:8888", // fiddler, watch if max tunnels reused
+		MaxConnectionPerHost: 4,
+		MaxIdleConnections:   10,
+		Timeout:              time.Second * 5,
+	}
+
```

Comment:
I don't mind adding these utilities to the golang side so things are more configurable. The question/issue is the implementation for this on the javascript side. Keep alive works this way on the golang side but for js the response won't be streamed, feel free to add these functions to the client file as added utilities. As long as this doesn't break any existing functionality I'll merge this in.